ddl/ddl_api.go

```diff
@@ -1139,6 +1140,12 @@ func getPrimaryKey(tblInfo *model.TableInfo) *model.IndexInfo {
 			allColNotNull := true
 			for _, idxCol := range key.Columns {
 				col := model.FindColumnInfo(tblInfo.Cols(), idxCol.Name.L)
+				// This Index has a column in DeleteOnly state,
+				// or it is expression index (it defined on a hidden column),
+				// it can not be implicit PK, go to next index iterator
+				if col == nil || col.Hidden {
+					continue IndicesLoop
```

Comment:
Could we use `break` instead of this line?

Reply:
No, if break here, L1149 will choose it as implicit PK.

Reply:
maybe sister's meaning: use a flag like `allColNotNull`, set it false, break, and check to skip in L1158

Reply:
> 
> 
> maybe sister's meaning: use a flag like `allColNotNull`, set it false, break, and check to skip in L1158

I think it is same, if there are not any performance and safety problem, I'd like keep it.

Reply:
it's ok

Reply:
I think if `IndicesLoop` isn't necessary, we'd better don't use it. PTAL @tiancaiamao

Reply:
Can we avoid the unconditional jump here?

> Go To Statement Considered Harmful

-- Edsger Dijkstra [[1]](https://en.wikipedia.org/wiki/Considered_harmful)

Reply:
Done